### ISSUE TITLE
Nano Adblocker / Defender for Firefox issues mega thread

### ISSUE THREAD

**Issue by jspenguin2017** [NanoAdblocker/NanoCore#41]
Warning: Someone submitted an unofficial build of the extension to the store, make sure to download the real one here: https://addons.mozilla.org/en-US/firefox/addon/real-nano-adblocker/

**Comment by Niftator** [comment]
The "get it for firefox" link at https://github.com/NanoAdblocker/NanoCore links here instead of, you know, anywhere useful, like an actual download link or anything.

**Comment by reunicorn** [comment]
@Niftator The extension is not yet available for download on Firefox. Right now it's only available for Chrome.

**Comment by jspenguin2017** [comment]
@Niftator I wrote "Work in progress", which means it's not done and ready yet.

**Comment by jspenguin2017** [comment]
@Niftator It is up in the store now, I think I packaged the right files. 
https://addons.mozilla.org/en-US/firefox/addon/real-nano-adblocker/

**Comment by nanisise** [comment]
Is it possible for supporting firefox 56?

**Comment by xlollomanx** [comment]
@nanisise I read somewhere jspenguin will not support ff56 because it need a different language to build extension and it's going to die.

**Comment by jspenguin2017** [comment]
For Defender, no, because I absolutely need the new API in FF 57. 
Nano Adblocker supports back to FF 52 like uBO.

**Comment by TonyTough** [comment]
Just a cosmetic issue:
If using Nano Defender with Ublock Origin, the additional filter lists Nano Base, Nano Timer and Nano Whitelist are shown as URL under addtional filters in Ublock Origin
whereas Nano Defender Integration List is shown correctly as list name.

**Comment by jspenguin2017** [comment]
OK, it's probably reading the `! Title:` header...

**Comment by jspenguin2017** [comment]
I can only debug one add-on at a time in Firefox? WTF???

**Comment by 88b88power** [comment]
Are there any websites where we can test whether the Nano Defender is working?

**Comment by jspenguin2017** [comment]
@88b88power https://www.cloudflare.com/apps/noadblock/install

**Comment by Chrisnetika** [comment]
Hi guys, I'm a new user though before I have been using UBlock.I have a question, I want to mod the popup thingie idk what it's call where you can disable ad blocker and has buttons where you can zap element.What I want to do is add a button where I can enable/disable my filters on the fly.Reason why? Well I only have one custom filter and it's for https://*-edge-chat.facebook.com and what that does is hide last active on facebook as well semi disables fb chat.What I basically want to do is toggle that on the fly.I figure bit coding hack that toggles my filters setting fast.Is this possible? I understand every update will revert this and I would have to manually edit it back in every update.

**Comment by elypter** [comment]
why not create a userscript?

**Comment by jspenguin2017** [comment]
@Chrisnetika You would need to fork, Chrome will enforce code integrity of extensions and once signed, it won't load if changed.

**Comment by Chrisnetika** [comment]
Or I could repackage it while in developer mode, then whitelist the extension ID using group policy templates problem solved.I was looking around files and pinpointed the file I would pro need to edit is js/popup.js.There already 4 buttons in tools at the bottom ie: toggle remote font,disable cosmetic filtering.. etc so i'm thinking can add one for my filters toggle.Only thing is... not sure what I need to add unless I spend hours looking through source and do trial and error lol.But yes I would need to fork ,do my edits and any updates I can just pull and preserve my edits, package the extension then add the extension i.d to whitelist via group policy (gets rid of annoying msgs in chrome)

**Comment by jspenguin2017** [comment]
@Chrisnetika You need to send a message to the background page, then: 
- Remove your filter from `nano.selectedFilterLists`
- Call `nano.loadFilterLists`

You should check to make sure `nano.selectedFilterListsLoaded` is `true` and `nano.loadingFilterLists` is `false` to prevent savedata corruption.

**Comment by Chrisnetika** [comment]
alright that a step in the right direction, thank you

**Comment by necros2k7** [comment]
Hi there. 1. Is it possible to use Nano defender with Umatrix, Ublock?
2. Can someone give cons & pros for using Nano blocker not other blockers?

**Comment by jspenguin2017** [comment]
1. Yes. See: https://jspenguin2017.github.io/uBlockProtector/
2. uBO is more stable and can be slightly faster, Nano goes for bleeding edge and have more features, but in turn is less stable. The performance hit for loading Nano's resources into uBO is untested, Nano will cache its extra resources efficiently.

**Comment by AmineHadef** [comment]
Works great on Firefox Quantum! Thanks.

**Comment by bobsage123** [comment]
Apologies, not a developer but have a question. If we use this for firefox and ublock, should we totally get rid of AAK-Cont? Or can they work in conjunction?

**Comment by Atavic** [comment]
Nanodefender, an [Anti-adblock defuser for Nano Adblocker](https://nanoadblocker.github.io/NanoDefender/) previously called uBlock Protector, is the current addon that supersedes AAK-Cont script.

**Comment by jspenguin2017** [comment]
You don't need AAK-Cont in that case.

**Comment by Francois-dag** [comment]
Not sure, but I guess that nano defender is involved within a loop process when using 1fichier.com.
Cheers.

**Comment by jspenguin2017** [comment]
@Francois-dag I can't reproduce. If you still have problems, please open another issue with more details and I'll have a look.

**Comment by Francois-dag** [comment]
https://1fichier.com/?2a5j7r9quz&af=127778

This one loops for me in the meantime :

- last firefox
- ghostery
- ublock-origin
- last nano defender

cheers.

**Comment by jspenguin2017** [comment]
Still can't reproduce.

**Comment by Francois-dag** [comment]
problem is on my side then, many thanx !

**Comment by nymous** [comment]
Hi,

NanoDefender addon is not found at this link https://addons.mozilla.org/en-US/firefox/addon/real-nano-defender/ (or when searching the AMO).

**Comment by jspenguin2017** [comment]
OK, Mozilla didn't like it, well, that's unfortunate.

**Comment by jspenguin2017** [comment]
This is the email I got: 

```
Hello,


Your add-on Nano Defender has been reviewed and one or more of of its versions have been disabled due to critical issues discovered.

Details:
This add-on is creating DOM nodes from HTML strings containing potentially unsanitized data, by assigning to innerHTML, jQuery.html, or through similar means. Aside from being inefficient, this is a major security risk. For more information, see https://developer.mozilla.org/en/XUL_School/DOM_Building_and_HTML_Insertion . Here are some examples that were discovered:
- content/1-content-domlib.js line 343
- content/1-content-domlib.js line 402
- content/1-content-domlib.js line 414
- content/1-content-domlib.js line 425
- content/1-content-domlib.js line 438
- content/3-content-rules-2-specific.js line 2588
- content/3-content-rules-2-specific.js line 2678
- content/3-content-rules-3-sticky.js line 24
- content/3-content-rules-4-proprietary.js line 33
- content/4-content-debug.js line 105
- content/4-content-debug.js line 132

A suggestion for the next version:
a) Please use node.textContent instead of innerHTML for assigning text information to a DOM node or if you need to create new DOM nodes use document.createElement() and NODE.appendChild() functions. Usually usage of innerHTML is not permitted in add-ons because innerHTML is a major security risk when used with data from external sources.
For more information, see https://developer.mozilla.org/en/XUL_School/DOM_Building_and_HTML_Insertion
For inserting text, textContent (or JQuery text) or createTextNode() should be used instead of innerHTML.
For inserting HTML, the safer method is to use createElement(), textContent, appendChild() instead of innerHTML.
b) Please use an established library for sanitize when creating DOM elements using external data. You can use DOMPurify, see accepted versions here: https://github.com/mozilla/amo-validator/blob/master/validator/testcases/hashes-allowed.txt

1) We don&#39;t allow add-ons to use remote scripts because they can create serious security vulnerabilities. We also need to review all add-on code, and this makes it much more difficult. Please insert those scripts locally from your add-on code.
- content/3-content-rules-4-proprietary.js line 66

Version(s) affected and disabled:
13.11, 13.15, 13.16, 13.17, 13.20, 13.24, 13.27, 13.28, 13.29, 13.30, 13.31

Reviewer:
Bogdan Matei

You can resolve this by addressing the issues raised in the reviewer's notes and uploading a new version at https://addons.mozilla.org/en-US/developers/addon/real-nano-defender/versions

If you want to respond please reply to this email or visit https://addons.mozilla.org/en-US/developers/addon/real-nano-defender/versions .

You can also join us in #addon-reviewers on irc.mozilla.org.

To learn more about the review process, please visit https://developer.mozilla.org/en-US/Add-ons/AMO/Policy/Reviews
--
Mozilla Add-ons Team
https://addons.mozilla.org


===================================================================
You received this email because you're subscribed to:
   * email me when my add-on is reviewed by a reviewer


You can't unsubscribe from this type of email.

Manage email notifications: https://addons.mozilla.org/users/edit#acct-notify
```

First, all innerHTML assignments are either hard-coded or generated from trusted data. Find a way to exploit it, I challenge you. 

Second, the proprietary code loader only run if debug mode is on, which is not the case for AMO version.

**Comment by Atavic** [comment]
@jspenguin2017 https://add0n.com/

That's a good place.

**Comment by jspenguin2017** [comment]
Even if I make it unlisted, it still needs to be signed by Mozilla. 
I have a server and I can host it myself, but that's not the problem.

**Comment by TonyTough** [comment]
Bug 1436849
Nano Defender is no longer on Add-ons Mozilla
https://bugzilla.mozilla.org/show_bug.cgi?id=1436849

**Comment by jspenguin2017** [comment]
![image](https://user-images.githubusercontent.com/7283682/36003176-1b867bfc-0cea-11e8-917f-749247cdbead.png)
Well, I guess they don't want to talk, I don't see how opening an issue will help. 

I'm still reviewing their questionable policy. If assigning to `innerHTML` counts as "Create or expose security vulnerabilities", well, I'm not spending time separating rules that use `innerHTML` and rules that do not. 

That apply to unlisted extensions as well, so there is no point spending time making it unlisted.

**Comment by elypter** [comment]
this whole signing enforcement thing was just a powergrab and a move towards google. i wish waterfox had more developers.

**Comment by jspenguin2017** [comment]
The problem with Waterfox is they are trying to keep legacy extensions, which is probably not going to go very well.

**Comment by elypter** [comment]
every deviation from the original project is going to take some work. i still hope mostly automatic merging tools. that would be a game changer for free software.

**Comment by dyn888** [comment]
Could you please upload 13.31 xpi somewhere? I have 'Nano Defender-13.30.xpi' downloaded, installed it without any problem, but now i see 13.31 was released, so it would be nice to be able to upgrade it manually, until things are resolved.

**Comment by jspenguin2017** [comment]
[Nano_Firefox_Defender.zip](https://github.com/NanoAdblocker/NanoCore/files/1711530/Nano_Firefox_Defender.zip)

**Comment by dyn888** [comment]
Thanks! I loaded it via extension debugging, albeit it can be a bit of a hassle to use it that way. Other that that, it works fine.

If anyone has original XPI (from Mozilla page), please upload. :)

**Comment by dyn888** [comment]
Just a quick question -- is the difference between 13.30 and 13.31 only in videolab.io being added?

BTW, Nano Defender also works with FF 55.x, tested with a few sites on Waterfox 55.2.2 and Basilisk (Firefox/55.0 Basilisk/20180202), so maybe min version required could be lowered, albeit there may be some reason to keep it at 57, just saying.

**Comment by jspenguin2017** [comment]
I bump the version for every change, so that's likely the case. 

It doesn't always work on Firefox 55, I can add more compile flags when platform building is working.

**Comment by setosd000** [comment]
I have nano defender 13.31 from the directory of my firefox 58.
[nanodefender-13-31.zip](https://github.com/NanoAdblocker/NanoCore/files/1712084/nanodefender-13-31.zip)

**Comment by MusicInMe** [comment]
Hi guys, sorry to bother yous, not familiar with coding. Can't install ver 13.31 on FF 56.0 (64 b) as it says, that the addon can not be installed because it "appears to be corrupt".

**Comment by dyn888** [comment]
Thank you very much, setosd000! :) I installed XPI on latest FF and it works 100%. For all interested, if you want to rename it, the original name from Mozilla site was: `nano_defender-13.31-an+fx.xpi` - according to Google Cache.

MusicInMe, you need to extract the zip first, however, it won't get installed because 13.31 has `"strict_min_version": "57.0"` in it, while 13.30 has `"minimum_chrome_version":"59.0"`, which is less strict and allows installation on older browsers.

I am uploading 13.30 for all those who want it. It was downloaded from Mozilla store. First unpack the zip and then drag&drop it to your browser. Note: all addons have all original files signed with Mozilla's private key, any change, intentional or not, would prevent it from being installed, so you can be sure it is safe to manually install it.
[nano_defender-13.30-an+fx.zip](https://github.com/NanoAdblocker/NanoCore/files/1712993/nano_defender-13.30-an.fx.zip)

**Comment by jspenguin2017** [comment]
That's not my Firefox build, someone used Foxified to sign the Chrome build. 

> so you can be sure it is safe to manually install it

As Mozilla sign everything and review later, it hardly has any security, especially for unlisted Foxified extensions.

**Comment by jspenguin2017** [comment]
I have some code that uses Firefox 57's response body filtering but it's guarded by Firefox platform flag instead of feature detection, it's not a great design, I have plan in fixing it but it requires so much changes that I haven't get to it. The original extension was hard coded for Chrome.

**Comment by dyn888** [comment]
So, original 13.30 also has strict_min_version? I will edit my post. Yes, Mozilla does that, sign first and review later, still, it is a good way to ensure integrity of something, at least when it comes from official source.

**Comment by DarkWav** [comment]
You could make Nano Defender a script for tampermonkey or something like that. Then Mozilla cant pervent
you from distributing it for irrelevant reasons. Bu thats still your choice and i know you had your reasons to discontinue AAK-cont.

**Comment by TonyTough** [comment]
Yes, but would it be possible to just "encapsulate" the Add-on in a script for tampermonkey without further changes?

**Comment by wazerstar** [comment]
I got latest version that was signed, grabbed it out from my cache. you can install it as drag and drop og locate file with firefox it will accept it.
[nano-defender-signed.zip](https://github.com/NanoAdblocker/NanoCore/files/1714252/nano-defender-signed.zip)

**Comment by krystian3w** [comment]
- [<sup>{6ea144f3-db99-47f4-9a1d-815e8b3944d1}</sup> **.xpi.txt**](https://github.com/NanoAdblocker/NanoCore/files/1714334/6ea144f3-db99-47f4-9a1d-815e8b3944d1.xpi.txt) - <ins>remove</ins> **".txt"** <sup>(recommend "s**ave as...**" option)</sup>

- [<sup>{6ea144f3-db99-47f4-9a1d-815e8b3944d1}</sup> **.zip**](https://github.com/NanoAdblocker/NanoCore/files/1714337/6ea144f3-db99-47f4-9a1d-815e8b3944d1.zip) - <ins>change</ins> **".zip"** to **".xpi"** <sup>(recommend "**save as...**" option)</sup>

PS. Possible install with ".xpi.txt"/".zip" when **drag & drop** file to "<ins>**about:addons**</ins>" page.

**Comment by krystian3w** [comment]
- [<sup>{6ea144f3-db99-47f4-9a1d-815e8b3944d1}</sup> **.xpi.txt**](https://github.com/NanoAdblocker/NanoCore/files/1714334/6ea144f3-db99-47f4-9a1d-815e8b3944d1.xpi.txt) - <ins>remove</ins> **".txt"** <sup>(recommend "**save as**..." option)</sup>

- [<sup>{6ea144f3-db99-47f4-9a1d-815e8b3944d1}</sup> **.zip**](https://github.com/NanoAdblocker/NanoCore/files/1714337/6ea144f3-db99-47f4-9a1d-815e8b3944d1.zip) - <ins>change</ins> **".zip"** to **".xpi"** <sup>(recommend "**save as**..." option)</sup>

PS. Possible install with ".xpi.txt"/ ".zip" when **drag & drop** file to "<ins>**about:addons**</ins>" page.

**Comment by jspenguin2017** [comment]
@DarkWav That's not possible. I need background page scripting.

**Comment by jspenguin2017** [comment]
I'll pass the extension though dead code eliminator after my exams. If Mozilla still doesn't like it then I'll go unlisted.

**Comment by dyn888** [comment]
That's great to hear! :)

Just for info, for new people who may wonder what to download. The following files are binary identical, that is, exactly the same file:
`{6ea144f3-db99-47f4-9a1d-815e8b3944d1}.xpi` (from `nanodefender-13-31.zip`, posted by **setosd000**)
`nano-defender-signed.xpi` (from `nano-defender.13.31-signed.zip`, posted by **wazerstar**)
`6ea144f3-db99-47f4-9a1d-815e8b3944d1.zip` (renamed xpi, posted by **krystian3w**)
`6ea144f3-db99-47f4-9a1d-815e8b3944d1.xpi.txt`  (renamed xpi, posted by **krystian3w**)
SHA-1: 045743f733e3ca89bce9778a871ad36271460674
Thanks to all who uploaded! :)

The original filename from the Mozilla store was: `nano_defender-13.31-an+fx.xpi`

So, download any of these, they are all the same. If you have Firefox older than v57 you may want to check foxified versions above.

**Comment by krystian3w** [comment]
[Extension_Signing#Unbranded_Builds](https://wiki.mozilla.org/Add-ons/Extension_Signing#Unbranded_Builds "https://wiki.mozilla.org/Add-ons/Extension_Signing#Unbranded_Builds ")  - test install older version https://github.com/NanoAdblocker/NanoCore/issues/41#issuecomment-364468978

**Comment by krystian3w** [comment]
<ul>

No works in **Unbranded** (stable 58.0.2), **Iceweasel** and **Nightly** (60).
<kbd>xpinstall.signatures.required</kbd> - <kbd>true</kbd>

Tested too method: "[<kbd>config.js</kbd> + <kbd>config-prefs.js</kbd> / <kbd>mozilla.cfg</kbd> + <kbd>autoconfig.js</kbd>](http://www.largrizzly.net/firefox.html)" - outdated

</ul>

---

Possible another compile unsigned addon (I can install "uBlock0.webext.xpi" - **unsigned** in Unbranned / Nightly with change value <kbd>xpinstall.signatures.required</kbd>)?

**Comment by GrPK** [comment]
So what is the problem if I download the one from the add-ons page on the first post? Cause I dont get it why I should import the add-ons

**Comment by jspenguin2017** [comment]
It's about Nano Defender.

**Comment by Fl4m3ThR0wEr** [comment]
@jspenguin2017 Lol, do they seriously say: (We dont allow add-ons to use remote scripts because they can create serious security vulnerabilities.  

So they say Nano Adblocker/Defender have Remote scripts? Lol Lol

**Comment by krystian3w** [comment]
Maybe Mozilla team do not have time follow changes in remote files 😰

**Comment by Corruptinglyneedful** [comment]
Firefox install missing for Nano Defender?  Is this working yet?  Using FF 56.  Just ditched UBO because I heard this was better for blocking anti adblock which I'm steadily growing tired of.

**Comment by jspenguin2017** [comment]
Yea, I haven't get to it. Having technical difficulties with weird Mozilla policy.

**Comment by Corruptinglyneedful** [comment]
Can I force install it?

**Comment by Tinyhaker** [comment]
Yep: download this file: https://www.sendspace.com/file/7lcysq
Drag and drop into your browser

**Comment by krystian3w** [comment]
Old signed 13.31 or newest compile another method to works in:
* [Firefox Unbranded](https://wiki.mozilla.org/Add-ons/Extension_Signing#Unbranded_Builds)
* Firefox Nightly (`xpinstall.signatures.required` - `FALSE`)
* [Iceweasel](https://sourceforge.net/projects/libportable/files/Iceweasel/Stable)

**Comment by krystian3w** [comment]
Try install addon: https://addons.mozilla.org/pl/firefox/addon/checkcompatibility2/ and after install Nano Defender 13.31 from XPI file.

**Comment by Gigobitto** [comment]
Im running Chrome version thanks to Chrome Store Foxified. Do you think I should downgrade instead to the proper Firefox version or is fine like this?

**Comment by jspenguin2017** [comment]
You can just change the manifest, there will be exceptions on the background page but shouldn't cause too much problems. 
I'm still fixing my tools... It can take a while combined with exams and other stuff...

**Comment by Corruptinglyneedful** [comment]
@jspenguin2017 I don't know what that means.  I am a simple user who is tired of anti-adblocking ways and it does not seem to work for Firefox 56.

**Comment by jspenguin2017** [comment]
Yea, it doesn't yet. I'll add support for older Firefox later.

**Comment by Corruptinglyneedful** [comment]
Is there nothing I can do for anti-adblock then?  I don't use Chrome and I don't use Firefox 57 yet.

**Comment by krystian3w** [comment]
If **checkcompatibilty2** and/or **Chrome Store Foxified** doesn't help - yes.

**Comment by Corruptinglyneedful** [comment]
@krystian3w I have the addon installed but no xpi file to install and I can't seem to find one.

**Comment by Tinyhaker** [comment]
@Corruptinglyneedful  https://www.sendspace.com/file/7lcysq here you are the xpi file. MD5 check CFE225F320AE7D411230FD84E87F0644

**Comment by Corruptinglyneedful** [comment]
OK, I seen that earlier but wasn't sure because it was just a random file link, could have been anything.  However, I'll trust that you didn't just make up that MD5 check file and the fact you took time to post it, means you care about security.

I will give this a go, thank you.

**Comment by Corruptinglyneedful** [comment]
OK, this seems to have installed properly.  Now is that all I need to do or is there some sort of other steps?  Really liking this ad blocker as a fork of UBO.  Got a bit tired of Gorhil's stuffy attitude and refusal to add features.  This essentially does the same thing, with a dev that's much more helpful.  I can support that.

**Comment by krystian3w** [comment]
http://onlinemd5.com

**Comment by Tinyhaker** [comment]
@Corruptinglyneedful  i think you don't need to do anything more xD

**Comment by Corruptinglyneedful** [comment]
Thanks guys, Firefox needs to piss off with preventing me from installing things I shouldn't.  I can accept risk if something breaks.  I don't need browsers to be my parent.

**Comment by Corruptinglyneedful** [comment]
Oh, 1 more minor question.  I tried to import my UBO settings to Nano and they don't seem to have worked, even after I commit the changes?  I'm having to re-build my white list from scratch.  Is this normal?

**Comment by Tinyhaker** [comment]
@Corruptinglyneedful delete all white list in nano and import white list from UBO again :D

**Comment by Corruptinglyneedful** [comment]
I had done that originally, still didn't carry over the dynamic rules for some reason.  /shrug

There are worse things.

**Comment by Tinyhaker** [comment]
@Corruptinglyneedful doing the manual way =)) ctrl + a ; ctrl + c ; ctrl + v from UBO to Nano ..........

**Comment by Corruptinglyneedful** [comment]
Lol that would be painful slow to do 1 line at a time, yes that would be worse.

Another question has cropped up.  The worst offender of anti-adblock and the whole reason for looking into it; Wikia.com is detecting I"m using an adblocker, even with Nano Defender on.  It's only when I unblock this domain does it piss off.  xpanama.net  What is this domain?  Also can we just talk about how disgusting wikia is for a minute.  There's over 25 domains on this page, over half are ad servers or tracking companies...

**Comment by Tinyhaker** [comment]
https://postimg.org/image/gkthdc8t7/ *.Wikia.com domain ?? no problem so far

**Comment by Corruptinglyneedful** [comment]
I hate to be a bother with this but I made a screen recording to show what I mean, images weren't cutting it.

It should help clear up the confusion.

You can view that here: https://gfycat.com/SoftKeyBigmouthbass

**Comment by Tinyhaker** [comment]
@Corruptinglyneedful  https://postimg.org/image/pxu2h0paj/ Mine don't have that url .....

**Comment by Corruptinglyneedful** [comment]
Here's the URL that I was on when I recorded that. http://projectsalt.wikia.com/wiki/ProjectSalt_Wiki

**Comment by Tinyhaker** [comment]
@Corruptinglyneedful  see my image ? same url

**Comment by Corruptinglyneedful** [comment]
I edited my post.

**Comment by Tinyhaker** [comment]
I enable the wikia services nothing happen ctrl + f5 the page

**Comment by Corruptinglyneedful** [comment]
That's really strange.  Try setting it up like I have mine in the video.  All those domains should be loading unless maybe only one shows up when you block one of the others.

**Comment by Tinyhaker** [comment]
https://youtu.be/_0f83GCBSsA try like this hmmmh

**Comment by Corruptinglyneedful** [comment]
I notice that I am blocking and allowing way more domains than you.  That could be it.  Try to set it like how I had mine in the gfycat.  I'm pretty sure it's adblock detector script hosted on a 3rd party domain.  I google the domain and I can't find much about it.

How do you get yours to show all the sub domains?

**Comment by Tinyhaker** [comment]
you click on the +all

**Comment by Corruptinglyneedful** [comment]
Thanks man, that helps with f uture blocking mwhahaha.

Oh, well look at this. adserver.xpanama.net and as soon as I block it, it breaks the page and throws that warning even if I whitelist the initial domain.

It has to be an anti-adblocker script.  There's no ads on the page.

**Comment by Tinyhaker** [comment]
if i click the blue turn off button reload the page i see the panama domain turn back on no way to see it

**Comment by Corruptinglyneedful** [comment]
That's pretty strange, yah?  Are you on Firefox or Chrome?  I'm on Firefox 56.  Might make the difference.

I think I stumbled on their adblocking script somehow.

**Comment by Tinyhaker** [comment]
I'm on Firefox 58.0.2 with some anti adblock script throught

**Comment by Corruptinglyneedful** [comment]
I got Nano Defender installed, you guys helped with that.  It is still showing up.

**Comment by Tinyhaker** [comment]
Oh right i remember lol nano defender only working well with firefox 57+ ... never try on 56

**Comment by Corruptinglyneedful** [comment]
I installed it with that addon check compatibility thing.  Before it wouldn't work.

**Comment by jspenguin2017** [comment]
Make sure to enable the integration filter if you installed unofficial build of Nano Defender, as the integration messaging won't work as expected. 
![image](https://user-images.githubusercontent.com/7283682/36386439-f998a080-1552-11e8-8210-923050e3c898.png)

**Comment by Corruptinglyneedful** [comment]
Yup, that's on and updated.  I wouldn't say I installed an unofficial build, I just shoved it into Firefox 56 with these guy's help.  I'm not using UBO.  I'm using Nano Adblocker and Nano Defender.

**Comment by Tinyhaker** [comment]
i think time to upgrade to newer firefox version ....

**Comment by jspenguin2017** [comment]
Do you have a test link for the problematic page?

**Comment by Corruptinglyneedful** [comment]
The test link is this one http://projectsalt.wikia.com/wiki/ProjectSalt_Wiki

The problem is this one https://gfycat.com/SoftKeyBigmouthbass

The conversation about this starts here https://github.com/NanoAdblocker/NanoCore/issues/41#issuecomment-366709480

Anyways, I hope all that helps.

**Comment by jspenguin2017** [comment]
Does `wikia.com##script:inject(abort-on-property-write.js, adonisHash)` works?

**Comment by Corruptinglyneedful** [comment]
It might but I don't know where to put that.  I'm assuming it's a rule of some sort.

**Comment by jspenguin2017** [comment]
Just put in My filters.

**Comment by Corruptinglyneedful** [comment]
OK, added that to my filters, blocked xpanama.net and same error.  So going with a resounding NO on that one.

W/E this domain is they are using, seems to be directly related to detecting adblockers.  It seems to do nothing else.  There are no ads on the page, not even text ads.

**Comment by jspenguin2017** [comment]
It doesn't connect to that domain for me. Probably something on your side.

**Comment by Corruptinglyneedful** [comment]
Hmm,  I don't have anything special.  I'm using default Nanoblocker install and settings, minus tick for advanced user.

If you've set your filters on that page to how I have mine and it still doesn't show up for you, I don't' know what to say that.

No malware or any issues here and nothing special to interfere with that page.  All I know is that when I block xpanama.net, breaks page.

**Comment by krystian3w** [comment]
You are from Panama or your ISP?

**Comment by Corruptinglyneedful** [comment]
No, I'm from USA.

**Comment by SBI-boy** [comment]
Are you going to add a whitelist option on this extension?

**Comment by krystian3w** [comment]
Nope.

**Comment by jspenguin2017** [comment]
@SBI-boy Nano Adblocker already has whitelist, it is not possible to implement that in Nano Defender as of now.

**Comment by martinaoo** [comment]
Hi Dev, what do you mean when will the  nanodefender available aggain in Add-ons Mozilla.
AprIl/May? 

Thanks In Advance

**Comment by jspenguin2017** [comment]
It will be available when I get to it.

**Comment by TonyTough** [comment]
Isn't it possible to make a Beta version and host it here on github like ublock origin?
Or has Mozilla also blocked the signing for Beta versions of Nano Defender?

**Comment by jspenguin2017** [comment]
Maybe it's possible but I don't want to bother.

**Comment by krystian3w** [comment]
@TonyTough: You can try compile unsigned addon to normal works (**without** use  debugging install  for any run browser again - `about:debugging`) in:

* Firefox Nightly 60 (`xpinstall.signatures.required` - **`FALSE`**)
* Firefox Unbranded (https://wiki.mozilla.org/Add-ons/Extension_Signing#Unbranded_Builds)
* Iceweasel (https://sourceforge.net/projects/libportable/files/Iceweasel/Stable)

Maybe install usigned version without comments "addon may/was corrupt" - maybe edit **manifest.json** enough when see error in Firefox console (Ctrl + J) - I cannot fix it.

**Comment by Corruptinglyneedful** [comment]
Nano Defender is required for anti adblock.

**Comment by seba2282** [comment]
Very big combinations. If we have installed nano adblocker, nano defender should be fully integrated in it. If different adblocker, nano defender should be as separetly extension, which we want add for protect our adblockers. And after it sentense will stay only which browser support nano defender. :)

**Comment by jspenguin2017** [comment]
@seba2282 Nano Defender is not configurable, so I can't just merge the two extensions.

**Comment by Corruptinglyneedful** [comment]
We need Nano Defender for both UBO and Nano itself for adblocking capability.  To my knowledge there is no other anti-adblocker out there.

**Comment by jspenguin2017** [comment]
In that case you can try Nightly or just go with Chrome. 

I can put up a new build but I can't get it signed. 
[Nano_Firefox_Defender.zip](https://github.com/NanoAdblocker/NanoCore/files/1782975/Nano_Firefox_Defender.zip)

**Comment by jspenguin2017** [comment]
@Corruptinglyneedful Please complain to Mozilla, who refused to put the extension on the official Firefox store.

**Comment by xlollomanx** [comment]
@Corruptinglyneedful I'm been able to find a workaround to get addon signed. I followed this guide https://developer.mozilla.org/en-US/Add-ons/Distribution/Submitting_an_add-on . When it ask "how to distribute this version" I choose "On your own", then select the zip file that @jspenguin2017 has provided here two post ago and then wait for process completion. It gives me an xpi file downloadble and now I can get it work on firefox 58.0.2. I don't know if I'm doing something wrong but it works for now.

**Comment by Aaron-Ritter-IDEXX** [comment]
@xlollomanx According to Mozilla's site, that sort of "unlisted addon" signed through the web site will be put through the same review process. I'm not sure if that's the best approach. Opinions on that from someone who knows how this works better than I do?

Pity that the Android version of Chrome doesn't support extensions. The only other mobile browser that does besides Firefox is Dolphin, and there's no Dolphin port of uBlock or Nano Adblocker, plus Dolphin has its own issues.

**Comment by xlollomanx** [comment]
@Aaron-Ritter-IDEXX Umh I didn't know, but from what I can read if I set "On your own" the addon should be keep private and not published.

**Comment by TonyTough** [comment]
> if I set "On your own" the addon should be keep private and not published on the store, but I don't know if it should pass a review process too.

If it isn't hosted on their site, they can't delete it like th current version.

So this way seems to seems to go in the direction of my suggestion a few posts above (to make a beta version and host it on github or somewhere else)

**Comment by jspenguin2017** [comment]
@Aaron-Ritter-IDEXX What about Yandex browser? Does its Android version has extensions? 
I think WebStore doesn't work with it so you may need to side load packages. 

Also, I think something else needs to be done for Android. You need to block ads from other apps too, and maybe SMS and stuff. Currently, open source system wide adblockers are [DNS66](https://github.com/julian-klode/dns66) and [Blokada](https://github.com/blokadaorg/blokada), but both rely on DNS and cannot handle more complex rule than blocking an entire domain. 

There isn't any functional open source TLS repacking libraries, which is kind of annoying. There are so many commercial solutions though... It's horrifyingly hard to make such library because the documentations are either terrifyingly complex or literally useless.

**Comment by Corruptinglyneedful** [comment]
It was my understanding they gave you the reasons they wanted you to fix so that they could put it in the store?

I refuse to move to any other browser that does not support Tab Mix Plus.

**Comment by jspenguin2017** [comment]
@Corruptinglyneedful Yes, but the reasons are ridiculous. I need to use `innerHTML` and DOMPurify is no-go. 

Also, I'm quite busy these days, and I have better things to do than dealing with Mozilla's nonsense.

**Comment by TonyTough** [comment]
> I'm been able to find a workaround to get addon signed. I followed this guide https://developer.mozilla.org/en-US/Add-ons/Distribution/Submitting_an_add-on . When it ask "how to distribute this version" I choose "On your own", then select the zip file that @jspenguin2017 has provided here two post ago and then wait for process completion. It gives me an xpi file downloadble and now I can get it work on firefox 58.0.2.

Can you post a download link to this current self-signed version?

**Comment by xlollomanx** [comment]
@TonyTough the link should be this [](https://addons.mozilla.org/firefox/downloads/file/884162/nano_defender-13.39-an+fx.xpi?src=devhub), but I don't know if anyone can download it.
 @jspenguin2017 about android ads blocking did you know Adaway [?](https://forum.xda-developers.com/showthread.php?t=2190753), it requires root but I'm using it from years and I think is a good alternatives. Otherwise @Aaron-Ritter-IDEXX  there are on android chromium based browser that have an integrated adblock like Brave Browser (on Play Store), or Bromite on xda [https://forum.xda-developers.com/android/apps-games/app-bromite-chromium-plus-adblocking-t3687258](url), they are not good like nano adblock & nano defender, but I think they are better than nothing.

**Comment by Aaron-Ritter-IDEXX** [comment]
This is a little bit off topic, but I'll answer the question.

Yandex is a little odd. It says you can install extensions from the Opera store, but I could only figure out how to install them from the Chrome Web Store. Go figure. As for the browser itself, it seems decent enough from a few minutes' use, but I can't find any benchmarks, so I suppose I'd have to run them myself.

Nano Adblocker installs and appears to function. It does not appear in the "Content blocker extension" menu, which is fine. I don't know why they have a separate "content blocker" API, and I wonder what limitations there may be on content blocking if you're not using it.

Nano Defender is marked as "Not Compatible". It isn't clear why.

As for blocking ads on Android in general, a quick search isn't revealing any present app that can do that properly and not just at the DNS level. AdBlock Plus used to have an app but seems to have discontinued it; no new builds since 2016. There is MinMinGuard, an Xposed module, which works by modifying applications not to display ads, if they use one of the libraries it knows how to patch. Not quite what I'd want to use. But that's okay. I'm mostly concerned with blocking web ads.

**Comment by Corruptinglyneedful** [comment]
Fuck Chrome, honestly.  I fucking despise that browser and I will not use it.  Their restrictive security bullshit hurts extensions and addons I use like again Tab Mix Plus, the absolute best addon ever and it hurts my ability to customize my browser.  That is why I'm using Firefox.  I refuse to move to anything that does not support Tab Mix Plus.

Hmm, if you don't have time to maintain the project perhaps it's time to fork it and let someone else take it over.

Nightly, Waterfox, ESR, etc are not official versions of Firefox and are not officially supported, also they reek hell with many extensions, scripts or addons any time something is updated.

They are not suitable replacements.

We need support for the official versions of Firefox for Nano Defender.

**Comment by bb010g** [comment]
Have you not been able to talk to someone about this on IRC (#addon-reviewers on Mozilla)? You might have more luck with an exception due to the heavier needs of an addon like this.

**Comment by TonyTough** [comment]
> @TonyTough the link should be this https://addons.mozilla.org/firefox/downloads/file/884162/nano_defender-13.39-an+fx.xpi?src=devhub , but I don't know if anyone can download it, if not work I'll attach the file here.

The link doesn't work.

**Comment by xlollomanx** [comment]
@TonyTough Ok I'll attach here. Remember to rename file from .txt to .xpi
[nano_defender-13.39-an+fx.txt](https://github.com/NanoAdblocker/NanoCore/files/1789308/nano_defender-13.39-an.fx.txt)

**Comment by Aaron-Ritter-IDEXX** [comment]
Oh, it must be using a different user-agent on the Chrome Web Store in order to fool it. If I try the user-agent I pasted in here in the desktop browser, it gives me a mobile version of the store that doesn't offer to let me install it. I don't know a way offhand to see what user-agent Yandex mobile is using. I'm also not sure whether the user-agent string is how the web page is deciding whether it's "compatible" or not. (I know little about extension development).

**Comment by Aaron-Ritter-IDEXX** [comment]
I can verify that the beta version of the Yandex mobile browser, which like Chrome allows the sideloading of unpacked extensions in developer mode, can successfully install Nano Defender copied from another computer. So whatever the hangup is, apparently either it's just on the store or it doesn't prevent loading but breaks something down the road that isn't obvious at first glance.

**Comment by Aaron-Ritter-IDEXX** [comment]
Got remote debugging working. Yandex is setting the UA to "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/54.0.2840.71 Safari/537.36", and 54 is definitely *below* 59. That explains it. Clearly they need to update their UA string. I don't see a way to override it, in the meantime.

**Comment by spvkgn** [comment]
@jspenguin2017 
I'd like to learn is it proper to get latest zip for Firefox 57+ like this?
```
git clone https://github.com/jspenguin2017/uBlockProtector
cd uBlockProtector
git archive -o ../Nano_Defender.zip HEAD:Extension\ Compiler/Extension/
```

**Comment by jspenguin2017** [comment]
@spvkgn No, you need to run the build script: 
```Bash
# Install dependencies
curl -sL https://deb.nodesource.com/setup_8.x | sudo -E bash -
sudo apt-get install -y nodejs
# Close repos
git clone --depth 1 https://github.com/jspenguin2017/uBlockProtector
git clone --depth 1 https://github.com/NanoAdblocker/NanoCore.git
# Build
mkdir ./NanoCore/dist/build
cd ./NanoCore/scripts
node build-defender.node.js --firefox --use-ubr-repository
```
I didn't test, if it doesn't work then it'll mostly be typos.

**Comment by Corruptinglyneedful** [comment]
If someone does fork the Firefox version, please let me know.

**Comment by bb010g** [comment]
I don't see why a fork would be required. The work can be done here by another collaborator. Long-term forking is just more fragmentation.

**Comment by spvkgn** [comment]
@jspenguin2017 Thanks, it works, I've tested on Ubuntu 17.10.

By the way, [web-ext sign](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/web-ext_command_reference#web-ext_sign) can help to get signed extension from script with no distribution listing created on AMO.

**Comment by koboltzz** [comment]
Sorry to ask .... but why there is a github 'release' page for NanoCore (=Nano Adblocker firefox zip) but **not** for NanoDefender? 
Wouldn't it be easier for everyone just to have a download possibility here?
I am stuck at version 13.31 since the only download method (mozilla addon page) is not possible at that moment.
The version 13.39 from @xlollomanx do not work for me on firefox nightly (windows).

**Comment by TonyTough** [comment]
The version from @xlollomanx is working for me on Firefox 58.0.2 64bit.
But you have to remove version 13.31 manually, otherwise you have it installed twice.

**Comment by xlollomanx** [comment]
This morning I download ubuntu bash from windows store and after read for a while I understand how to run the script (that jspenguin posted) and build it (I'm building??first time ever I'm using linux, except on android). After I have to found where the file was builded, and the signed in mozilla, I hope I have done all correctly, but I'm testing it and seems it works. I'llpost download link here: https://www.dropbox.com/s/2upr9hww93vsx9a/nano_defender-13.41-an%2Bfx.xpi?dl=0

**Comment by TonyTough** [comment]
> If you have installed my previous "release" I think you can install this without uninstalling it first because the signature should be the same.

I confirm that it works.

**Comment by Francois-dag** [comment]
>> Version 13.39 from @xlollomanx is working for me on Firefox 58.0.2 64bit.

Yes, thanx a lot !

**Comment by ScottRFrost** [comment]
@xlollomanx 's 13.41 linked 3 above this post works for me.  Firefox Dev Edition 60.0b1 (64-bit) on Windows 10.

**Comment by jspenguin2017** [comment]
@xlollomanx The build tools do run natively on Windows, I usually build on Windows actually. You just need to install Node.js from https://nodejs.org/en/ , create the build output directory and double click the proper batch file.

**Comment by xlollomanx** [comment]
@jspenguin2017 thank you for your help, but get script working on ubuntu in a  windows 10 bash it's a big enough for me today. Otherwise I tried to understand what you writing in your previous post but I don't know how to use that tool, I'm not a dev, I don't know any informatic language (c++, java, etc), so I need a while to figure it out. In the meanwhile I use the script posted previously to make a firefox addon that I'll post here (if I not disturb), until you find how to get around mozilla policy or another dev help you on this side.

**Comment by troysjanda** [comment]
> This morning I download ubuntu bash from windows store and after read for a while I understand how to run the script (that jspenguin posted) and build it (I'm building??first time ever I'm using linux, except on android). After I have to found where the file was builded (understand where windows put ubuntu and his files require a while), and then I signed it in mozilla. I hope I have done all correctly, but I'm testing it and seems it works. This is the download link if anyone else want to test it: https://www.dropbox.com/s/2upr9hww93vsx9a/nano_defender-13.41-an%2Bfx.xpi?dl=0
> 
> If you have installed my previous "release" I think you can install this without uninstalling it first because the signature should be the same. Big thanks to jspenguin for his works!

confirmed working in 59.0 RC without issues.

**Comment by jshir** [comment]
Hi, I'm new to Nano after dropping uBlock since AAK was killed off. 

I am trying to make clipconverter.cc stop opening popups to g3j2wzmon8b.com on click anywhere, using an obfuscated inline script near the bottom of the page after a like.php iframe. Other inline scripts are required to use the site. I tried ||clipconverter.cc##^script:has-text(window) since window prominently appears in the obfuscated script, but nope!

(Also - the Nano Filter Editor cannot copy text out of the filter list, which was rather annoying.)

[Nano] Nano Defender Activated :: www.clipconverter.cc
2-content-core.js:60:5
Loading failed for the <script> with source “https://defpush.com/ntfc.php?p=1422036”.
www.clipconverter.cc:29
Loading failed for the <script> with source “https://go.pub2srv.com/apu.php?zoneid=801499”.
www.clipconverter.cc:610
Loading failed for the <script> with source “https://g3j2wzmon8b.com/829548/”.
www.clipconverter.cc:1
[Nano] Nano Defender Activated :: www.facebook.com
2-content-core.js:60:5
[Nano] Excluded :: All Generically Applied Solutions

NA: v1.0.0.34 ND: 13.41
FF 58.0.2

**Comment by jspenguin2017** [comment]
Due to upstream bugs, I disabled HTML filtering by default. 
`clipconverter.cc##script:inject(abort-on-property-read.js, T0P5e)` works.

**Comment by jspenguin2017** [comment]
@bb010g I prefer a fork so I can remove Firefox specific code from Chrome version. Currently the compatibility shim is evaluated at runtime.

**Comment by Aaron-Ritter-IDEXX** [comment]
The Yandex feedback address replied, "The user agent will be necessarily updated with the matter of time. Thank you for your notion!". Hopefully I was clear that I meant the CWS override user-agent and not the regular user-agent. I tried, but there's a little bit of a translation barrier. Time will tell.

**Comment by xlollomanx** [comment]
update: https://www.dropbox.com/s/x81ort5le1g9zcb/nano_defender-13.42-an%2Bfx.xpi?dl=0

**Comment by Corruptinglyneedful** [comment]
What is that.

**Comment by troysjanda** [comment]
> update: https://www.dropbox.com/s/x81ort5le1g9zcb/nano_defender-13.42-an%2Bfx.xpi?dl=0

Changelog?

**Comment by xlollomanx** [comment]
@troysjanda just updated from jspenguin source

**Comment by jspenguin2017** [comment]
A style fix for Chromium 65 (only), I hope you tested it...

**Comment by xlollomanx** [comment]
@jspenguin2017 yes, it's working fine on my side. tested with two different pc on latest stable firefox

**Comment by ScottRFrost** [comment]
Got a build for 13.43, please?  Big performance fix for large arrays: https://github.com/NanoAdblocker/NanoFilters/issues/44#issuecomment-372459139

**Comment by xlollomanx** [comment]
@ScottRFrost Here you go: https://www.dropbox.com/s/wybh9q5xsct9y2a/nano_defender-13.43-an%2Bfx.xpi?dl=0 . Btw I'm switched from ff to chrome, I found a workaround to remove some things that I hate in Chrome. I'll use chrome for a while and test it.

**Comment by ScottRFrost** [comment]
Oh no! I don't want to get into a browser flame war, but please search the web for 'Edward Snowden NSA Google' or Wikipedia for 'PRISM (surveillance program)' before you downgrade from privacy-friendly fully-open-source Firefox to NSA-friendly Google Chrome with many, many parts closed source.  There's so much more at stake than just a few features!

**Comment by Corruptinglyneedful** [comment]
Look I don't care about all that.  I just want the thing working on Firefox 56 and up.

**Comment by Aaron-Ritter-IDEXX** [comment]
If you don't want to get into a browser flame war, the easiest way is not to start one.

**Comment by FaySmash** [comment]
@xlollomanx [OT] use chromium or Iron at least..

**Comment by jspenguin2017** [comment]
If you're using Windows, you probably have more to worry about; if not, you're probably using Chromium. I don't care which browser you use but I can choose which browser I support. If your browser is not on my supported list, then go ahead and fork it instead of trying to convince other people that the browser you use is better.

**Comment by SeriousHoax** [comment]
With the latest 13.43 version, Nano defender is working even after all these filters are disabled in Firefox. So, don't I need this anymore? Just installing the Nando defender extension will do the job?

![image](https://user-images.githubusercontent.com/29574622/37554624-acdfdfc0-2a05-11e8-9021-2a6ccf7d2f1a.png)

**Comment by Corruptinglyneedful** [comment]
Does it work on Firefox 56 @SeriousHoax

**Comment by FaySmash** [comment]
@Corruptinglyneedful it works for me

**Comment by jspenguin2017** [comment]
@SeriousHoax You definitely should have them enabled because otherwise Nano Defender may redirect analytics callback expecting the integration filter to block them.

**Comment by Corruptinglyneedful** [comment]
How does do?  Last time I tried to force install Nano Defender linked in
here, all firefox did was bitch it was incompatible and it locked up
Youtube's player so I had to yank it out.

**Comment by FaySmash** [comment]
@Corruptinglyneedful I disabled all compatibility/security/performance checks in waterfox

**Comment by SeriousHoax** [comment]
@jspenguin2017 Ok will do then. But there used to be a website or something explaining how to add them in Firefox with uBlock Origin as the Adblocker, from where I could add those filters to my uBlock Origin. I can't seem to find that page anymore. Wanna add those to my Firefox Android.

**Comment by jspenguin2017** [comment]
I'm not sure if it works on Android.

**Comment by SeriousHoax** [comment]
@jspenguin2017 Thanks. I don't see any "turn off your adblocker" type of warning in those sites so it's doing the job on Android too :)

**Comment by wazerstar** [comment]
Latest update to adblocker 1.0.0.36 breaks blocking things via gui.

Element mode picker not working
Right clicking  and blocking object not working.

**Comment by jspenguin2017** [comment]
Can't reproduce.

**Comment by wazerstar** [comment]
@jspenguin2017 

Forgot to say you need to have already blocked some content on the website.

Example I had this

! 3/5/2018, 2:41:32 PM https://www.faceit.com/en/client
www.faceit.com##.flex-center-start.main-header__message-bar__content

After removing it I could use things again.

-------

Another issue, adblocker picking does not work well with stylus addon.

Example, Have stylus addon installed and using a darksearch google theme, (tested many different all the same)

This is how it will show up.

http://prntscr.com/ismz31

**Comment by jspenguin2017** [comment]
@wazerstar 
Still can't reproduce. 
I won't fix conflicts with other extensions, it's up to you to choose what to use.

**Comment by Corruptinglyneedful** [comment]
Waterfox and other such variants are not Firefox, they are forks.

**Comment by wazerstar** [comment]
@Corruptinglyneedful ??? I'm using firefox quantum what are you talking about ???

**Comment by SBI-boy** [comment]
Hello, I'm using the latest verison on FF quantum, and it's causing trouble with ouo.io links... it is eithier causing a redirection to the root domain or a reload loop.

example link: http://ouo.io/go/e43ecx

please fix it


PS: this also happens in Opera

**Comment by jspenguin2017** [comment]
Can't reproduce.

**Comment by krystian3w** [comment]
> it's having conflict with ouo.io links... it is eithier causing a redirection to the root domain or a reload loop

I have loop in Firefox 59.0.1 + NA 1.0.36 + ND 13.43 (compiled/signed by @xlollomanx).

Text see in frame:
```
EAuykxGRRnwaEegFapn51CZk5hcwHMsFYx02Ia7j
```

> ![obraz](https://user-images.githubusercontent.com/35370833/37569448-adf21a1c-2ae2-11e8-96c0-8d3eda5a30f2.png) |
> --- |

**Comment by uwx** [comment]
I have the same issue with ouo.io, with the text box as well.

**Comment by krystian3w** [comment]
Maybe bug when cut `ouo.io` in `ouo.io` - (`http://ouo.io/go/e43ecx` reddirect should to `http://ouo.io`).

Short link to github works almost fine http://ouo.io/Py4r7F (see small box with text only once).

**Comment by Tinyhaker** [comment]
Anyone have 13.46 for firefox 59.01 signed version ?

**Comment by Tinyhaker** [comment]
Ah ok install just fine now. Just rename to Nano.xpi and then install just fine the 13.46 ^^

**Comment by xlollomanx** [comment]
if anyone need it this is my 13.46: https://www.dropbox.com/home/Nano%20Defender?preview=nano_defender-13.46-an%2Bfx.xpi

**Comment by Francois-dag** [comment]
>> if anyone need it this is my 13.46

Yes, but not found ...

**Comment by Corruptinglyneedful** [comment]
We need Nano Defender for Firefox 56-current version.

**Comment by Corruptinglyneedful** [comment]
Need Firefox 56-current please for Nano Defender.

**Comment by xlollomanx** [comment]
sorry @Francois-dag @Corruptinglyneedful , my fail now link should be fixed: https://www.dropbox.com/s/7s51569vy1mmkax/nano_defender-13.46-an%2Bfx.xpi?dl=0

**Comment by Francois-dag** [comment]
thanx @xlollomanx

**Comment by xlollomanx** [comment]
13.47: https://www.dropbox.com/s/zhwy51yofetcfcf/nano_defender-13.47-an%2Bfx.xpi?dl=0

**Comment by lalovzg** [comment]
@SBI-boy http://ouo.io/go/e43ecx still goes to the uou.io root (ND 13.46)
mmmmmm..... http://ouo.io/e43ecx OTOH.... (thanks i guess?) is working properly

**Comment by krystian3w** [comment]
Short links as <code>h<u></u>ttp://ouo.io/<strong>go</strong>/...</code> is no exist, maybe @SBI-boy</a> no have better links as NFSW contents.

Rddirect problem <code>h<u></u>ttp://ouo.io/<strong>go</strong>/...</code> to root  - is not triggered by ND + NA.

**Comment by Tinyhaker** [comment]
Anyone have 13.48 build ?

**Comment by krystian3w** [comment]
- [nano_defender-13.48-an+fx.zip](https://github.com/NanoAdblocker/NanoCore/files/1845457/nano_defender-13.48-an.fx.zip "github host") - drag and drop to Addons (`about:addons`) page.
- [nano_defender-13.48-an+fx.xpi](https://www.dropbox.com/s/4j5uli4k61nomac/nano_defender-13.48-an%2Bfx.xpi?dl=0 "dropbox host") - drag and drop to Addons (`about:addons`) page.

No override version by `@`**jspenguin2017**  / `@`**xlollomanx**. If does not work wait for version by `@`**xlollomanx**.

**Comment by troysjanda** [comment]
How many builds are there? are @krystian3w and @xlollomanx builds the same? Reason I ask is this could get very diluted with the possibility of many many builds by many many different maintainers, In my opinion there should be a single maintainer building and posting of XPI's. I am just thinking long term if everyone posts a signed version of there own can we be certain that there is no nefarious code being introduced speaking from a security standpoint. I have installed  the @xlollomanx build and I will continue using his or maybe consolidate changes into a single build. Just my opinion not trying to start trouble.

**Comment by krystian3w** [comment]
Mozilla no  recommended share `api-secret` key.  When I diff `@`**xlollomanx** file, no find other diffrences as in `common.js` file (~10 last lines) and lines minimal version Firefox:

<details><summary>common.js from <code>@</code><strong>xlollomanx</strong> 13.47 version</summary>

```
/**
 * The common bootstrapper for background and content script.
 */
"use strict";

/**
 * The main namespace. Different methods will be added to this namespace depending on
 * the execution context.
 * @var {Namespace}
 */
var a = {};

/**
 * Whether current browser is Firefox.
 * @const {boolean}
 */
a.isFirefox = false;
/**
 * Whether current browser is Edge.
 * @const {boolean}
 */
a.isEdge = false;

//@pragma-if-debug
/**
 * Whether the extension is loaded in debug mode.
 * The compiler is responsible in hard coding this switch.
 * @const {boolean}
 */
a.debugMode = true;
//@pragma-end-if

a.isFirefox = true;
a.debugMode = false;
```

</details>
<br>

<ins>So should be same</ins>. If not experience bug with 13.47 version - fixed in 13.48 (by `@`**jspenguin2017** - version for Chrome) you should wait to `@`**xlollomanx**.

**Comment by Tinyhaker** [comment]
Thanks you @krystian3w  your build working just fine ^^

**Comment by Corruptinglyneedful** [comment]
OK I don't understand all this developer talk and such.  I'm only here for Nano defender to be used on Mozilla Firefox v56 through the current version because it is not available on AMO, which to my understanding is their fault.

I require ND for Nano Adblocker.

**Comment by SeriousHoax** [comment]
@Corruptinglyneedful Download Nano defender from the dropbox link above. After downloading open your Add-ons menu in Firefox and drag the extension file there and install it.

**Comment by krystian3w** [comment]
This is mabe not enough. In the lastest version added minimal version to install addon:

```js
"applications": {
    "gecko": {
      "strict_min_version": "57.0"
    }
  }
```

In older post I suggested use [Disable Addon Compatibility Checks (Version 4-56)](https://addons.mozilla.org/pl/firefox/addon/checkcompatibility2/) to enforce  install (this is no fix lack in WebExtension API / Javascript Firefox 56).

**Comment by xlollomanx** [comment]
@troysjanda @krystian3w No, I didn't make any changes from jspenguin source. I recently learned how compile this addon using the script jspenguin provide some post ago. I don't have enough knowledge to modify it. I'll continue to compile nano defender time to time until jspenguin found another dev or remove firefox code from his source. I'm using chrome for a while, I still prefer firefox but I like more ads blocked than firefox, I'll leave it completely when jspenguin drop his support.

Btw new update 13.49, not tested just compiled:  https://www.dropbox.com/s/1h6rlmk3hlr4r4t/nano_defender-13.49-an%2Bfx.xpi?dl=0

**Comment by troysjanda** [comment]
@xlollomanx @krystian3w I wanted to clarify my previous post was meant only from a dev point of view and not because there was any nefarious code included. I was just saying that if there are multiple people providing builds, no one will know which build to install and the pool can get diluted.  Thank you both for making updates.

**Comment by xlollomanx** [comment]
@troysjanda  Yes, I understand your point of view. I don't know if it's possible share a common key...maybe creating new account on firefox and share with both of us. So who firstly see an update can update it.

**Comment by krystian3w** [comment]
No bad idea, I will find better method compile as :download to zip", to local disk (I have not figured out how to imorte only some of the changes are git 😋).

**Comment by Corruptinglyneedful** [comment]
OK now I'm confused.  I get how to install it now but  who has the updated and currently worked on version?  There's a lot of information and links in this discussion and a agree with the other users.  It's hard to follow what's going on.

Note, I'm in no way complaining about others working on it, I very much appreciate it.  I just want to make sure I get the correct version, to avoid compatibility issues.

Thank you for all your hardwork, folks!

**Comment by troysjanda** [comment]
@Corruptinglyneedful, Both @xlollomanx @krystian3w provide updated builds and offer links to those builds. @xlollomanx build will update over the previous jspenguin2017 build, however @krystian3w build will install alongside jspenguin2017 build and not overwrite it, showing as 2 Nano Defender addons in Firefox about:addons page.

I know this seems a bit confusing but it has to be this way due to the issue with jspenguin2017 build and Firefox AMO site. Which is my point in my earlier post about multi providers. Personally I use @xlollomanx build because it updates the currently installed Nano Defender and not add another build. 

I also think that the recent updates were addressing problems with the Chrome version.

**Comment by FaySmash** [comment]
@Corruptinglyneedful 13.49 is the most up-to-date version (as the version number suggests). I don't know what you mean with "currently worked on version", all here posted .xpi files are just compiled Addons of the current repo and there does only one exists for both chrome and firefox at the moment (though firefox support will be dropped in the future).
For "to avoid compatibility issues", all here posted .xpi Addons are compatible to each other, but if they're signed with a different dev key, you have to remove the previous version first (or you'll have the Addon installed twice). Also there **will be** compatibility issues with firefox 56 (although I haven't noticed anything  serious until now

**Comment by Corruptinglyneedful** [comment]
@FaySmash it is my understanding it is already dropped and now some others are working on it and fixing things.  See, it's too hard to tell what's going on.  I didn't even realize there were Chrome updates.

**Comment by xlollomanx** [comment]
@krystian3w i created new account on firefox, tell me where I can contact you to send login information

**Comment by krystian3w** [comment]
Maybe mail `nano [at] defender [dot] hub [dot] pl` (without spaces)?

**Comment by xlollomanx** [comment]
@krystian3w I send you account details, I hope I wrote your mail right :) 

Btw I resigned 13.49 with this new firefox key, so in the future anyone can install any version provided from krystian3w or me without any sign problems because they share the same key.
Before install it be sure to remove any other version of nano defender. 
https://www.dropbox.com/s/9lro1fmwk2bmkcc/nano_defender-13.49-an%2Bfx.xpi?dl=0

I hosted it on my dropbox but maybe we can find a better host for future, anyone have better option?

**Comment by jspenguin2017** [comment]
Can you guys start a collaboration in a real repository? This thread is getting unusable. 
Check what gorhill did to get auto-update working.

**Comment by krisu5** [comment]
Yeah, seriously. What's the point of posting these here, when you can make a fork instead.

**Comment by Corruptinglyneedful** [comment]
Keep us updated on what you are doing please.  I want to ensure we all get the updates.

**Comment by krystian3w** [comment]
> Keep us updated on what you are doing please. I want to ensure we all get the updates.

Really?

<details><summary> How I create self-signed Nano Defender: (click to read)</summary><br>

1.  Download latest uBlock Protector repository
2. Edit file manifest.json to  add verification minimum Firefox version and remove line minimal support chrome version
3. When I installed `node.js` and webxt program (to singing addon from console) run .bat"
    - I run Update `Compile.bat` > `uBO-Extra.bat` > `Disable Debug Mode.bat` **(Maybe unnecessary / dangerous)**
    - I run my `webxt-sign.bat`:
         ```
         web-ext sign --api-key=user:XXXXXXXXXXX --api-secret=XXXXXXXX --id={XXXXXXXXX} --source-dir=Extension --artifacts-dir=Signed
         ```
6. after a while time, I have self-signed an add-on in `Signed` catalog
</details>

---

> Can you guys start a collaboration in a real repository? This thread is getting unusable.
> Check what gorhill did to get auto-update working.

https://github.com/NanoAdblocker/NanoDefender/issues/33 maybe  waiting for  vrackfall  😄 

> I hosted it on my dropbox but maybe we can find a better host for future, anyone have any idea?

Maybe github/gitlab page / github/gitlab releases.

**Comment by Corruptinglyneedful** [comment]
OK very lost.

**Comment by krystian3w** [comment]
https://github.com/krystian3w/NanoDefenderFirefox/releases

Still I cannot merge only changes and automatic this process + create automatic self-signed version.

Maybe latter open [issues](/NanoDefenderFirefox/issues).

**Comment by jspenguin2017** [comment]
@krystian3w Check what gorhill did to make his self-signed version auto-update.

**Comment by Cs1cs0** [comment]
https://www.techradar.com beat it the Nano Adblocker +Nano Defender combo (windows 10 Firefox x64 59.0.2)

**Comment by jspenguin2017** [comment]
@Cs1cs0 Works fine for me.

**Comment by Francois-dag** [comment]
Sounds strange, a month ago I've reported a problem with 1fichier.com looping for me when Nano and co are enabled and Hugo said like in this case (works fine for me). So I guess it's a problem on a filtering list side. Well, my looping problem is still alive, currently I only use Chrome when 1fichier.com is requested !

**Comment by krystian3w** [comment]
Report in new issue if can reproduce (your configuration uBO/Nano + step by step) in Chrome.

**Comment by Cs1cs0** [comment]
Don't work

**Comment by aasril** [comment]
I also have the same problem with techradar. they detected adblocker (in my case uBO + Nano Defender) and redirect to this page https://www.techradar.com/cc.html. I'm on LinuxMint with Firefox 60.0b11.

**Comment by aasril** [comment]
@krystian3w I reinstalled Nano Defender with release from your page (v. 13.57). Followed all extra steps from https://jspenguin2017.github.io/uBlockProtector/. It resulted 3 extra filters on uBlockOrigin filters list: NanoBase, NanoDefender, and NanoWhitelist. Is that what final result should be like?. I also have added https://jspenguin.com/NanoAdblocker/AssetsMirror/NanoResources.txt to userResourcesLocation.

The reason I ask is because previously I have another filter named Nano Defender Integration in custom filter segment of UBO which I had removed prior to reinstall.

**Comment by Corruptinglyneedful** [comment]
Because Nano Defender no longer works with Firefox and they don't care to fix it.  They discontinued the project and care to neither support version 56 and beyond.

It is ultimately disappointing to say the least.

I refuse to upgrade Firefox until TMP supports the new web extension API.

**Comment by Corruptinglyneedful** [comment]
Only wants to support memory hog chrome which scarified user options and customization for security rather than work with Mozilla and get the problem sorted out.

**Comment by krystian3w** [comment]
@aasril I can read articles <ins>techradar.com</ins>, see "Recommended videos for you..." and no have redirect.
_(Firefox 59.0.2 x86, Windows 10 Pro x86)_

Maybe bug with specific list (non-default) or exclusion GNU/Linux (try override user-agent to Windows).

**Comment by aasril** [comment]
@krystian3w changing user agent to any desktop browsers doesn't work but changing it to mobile works (Android-Chrome).

**Comment by vietzerg** [comment]
@aasril Don't know if you tried blocking `inline scripts` yet. That method got rid of that redirection for me, but the majority of images on that website also were blocked.

**Comment by jspenguin2017** [comment]
It is whitelisted in Nano Defender, and it's intentional.

**Comment by mzso** [comment]
[**@FaySmash**](https://github.com/FaySmash) commented on [2018. márc. 27. 07:27 CEST](https://github.com/NanoAdblocker/NanoCore/issues/41#issuecomment-376401381 "2018-03-27T05:27:02Z - Replied by Github Reply Comments"):
>  (though firefox support will be dropped in the future).  
> 

Any particular reason why the FF version is being dropped?

**Comment by jspenguin2017** [comment]
@mzso The UserCSS code is conflicting with Chrome version.

**Comment by elypter** [comment]
could you at least keep the old version? i wanted to try out something with a new profile and was searching like crazy because it was still there a few days ago.

**Comment by jspenguin2017** [comment]
The support for Firefox has officially ended. No more Firefox build will be produced.

**Comment by jspenguin2017** [comment]
@elypter Probably not, the latest version has a regression on Firefox.
